Sources/Workspace/Workspace.swift

```diff
@@ -1885,16 +1887,29 @@ extension Workspace {
                     forceResolution: forceResolution,
                     extraConstraints: extraConstraints,
                     diagnostics: diagnostics,
-                    retryOnPackagePathMismatch: false
+                    retryOnPackagePathMismatch: false,
+                    removeResolvedFile: removeResolvedFile
                 )
-            } else {
+            } else if removeResolvedFile, self.fileSystem.exists(self.resolvedFile) {
                 // If we weren't able to resolve properly even after a retry, it
                 // could mean that the dependency at fault has a different
                 // version of the manifest file which contains dependencies that
                 // have also changed their package references.
-
-                diagnostics.emit(error: "the Package.resolved file is most likely severely out-of-date and is preventing correct resolution; delete the resolved file and try again")
```

Comment:
@abertelrud @neonichu @aciidb0mb3r the original error message suggests that such resolution exhaustion could occur if the Package.resolved file is far behind which is why I added the logic to try and delete it + retry. Do you think this is still a valid use case, or was that here for historical reasons? If it is still valid, can you think of a good way to create a test with such Package.resolved file?

Reply:
I somewhat suspect that it mostly became a bogus error message after switching to pubgrub if you're not hitting a test failure after making this change...

Reply:
Yes, my impression of this from bug reports is that this is a fallback "something went wrong" error message that doesn't actually relate to Package.resolved.  I don't know the history but at this point it seems to do more harm than good, so removing it is correct.